test/gomod.collector.test.ts

```diff
@@ -235,15 +294,108 @@ github.com/pierrec/lz4`);
+    });
+    expect(deps[1]).is.eql({
+      name: { value: 'github.com/pierrec/lz4', position: { line: 0, column: 0 } },
+      version: { value: 'v2.5.3', position: { line: 14, column: 92 } }
+    });
+    expect(deps[2]).is.eql({
+      name: { value: 'github.com/davecgh/go-spew', position: { line: 0, column: 0 } },
+      version: { value: 'v1.1.2', position: { line: 15, column: 74 } }
+    });
+    expect(deps[3]).is.eql({
+      name: { value: 'github.com/davecgh/go-spew/spew@github.com/davecgh/go-spew', position: { line: 0, column: 0 } },
+      version: { value: 'v1.1.2', position: { line: 15, column: 74 } }
+    });
+    expect(deps[4]).is.eql({
+      name: { value: 'github.com/pmezard/go-difflib', position: { line: 0, column: 0 } },
+      version: { value: 'v1.3.0', position: { line: 11, column: 41 } }
+    });
+  });
+
+  it('tests replace block before require in go.mod', async () => {
+    fake(`go list -f '{{ join .Imports "\\n" }}' ./...`, `github.com/alecthomas/units
+github.com/pierrec/lz4
+github.com/davecgh/go-spew
+github.com/davecgh/go-spew/spew
+github.com/pmezard/go-difflib`);
+    const deps = await collector.collect(`
+        module github.com/alecthomas/kingpin
+        go 1.13
+
+        replace (
+          github.com/alecthomas/units => github.com/test-user/units v13.3.2
+          github.com/pierrec/lz4 v2.5.2-alpha+incompatible => github.com/pierrec/lz4 v2.5.3
+          github.com/davecgh/go-spew => github.com/davecgh/go-spew v1.1.2
+          // replace github.com/pmezard/go-difflib v1.3.0 => github.com/pmezard/go-difflib v1.3.1
+        )
+
+        require (
+          github.com/alecthomas/units v0.1.3-alpha
+          github.com/pierrec/lz4 v2.5.2-alpha+incompatible
+          github.com/davecgh/go-spew v1.1.1+incompatible
+          github.com/pmezard/go-difflib v1.3.0
+        )
+      `);
+    expect(deps.length).equal(5);
+    expect(deps[0]).is.eql({
+      name: { value: 'github.com/test-user/units', position: { line: 0, column: 0 } },
+      version: { value: 'v13.3.2', position: { line: 6, column: 69 } }
+    });
+    expect(deps[1]).is.eql({
+      name: { value: 'github.com/pierrec/lz4', position: { line: 0, column: 0 } },
+      version: { value: 'v2.5.3', position: { line: 7, column: 86 } }
+    });
+    expect(deps[2]).is.eql({
+      name: { value: 'github.com/davecgh/go-spew', position: { line: 0, column: 0 } },
+      version: { value: 'v1.1.2', position: { line: 8, column: 68 } }
+    });
+    expect(deps[3]).is.eql({
+      name: { value: 'github.com/davecgh/go-spew/spew@github.com/davecgh/go-spew', position: { line: 0, column: 0 } },
+      version: { value: 'v1.1.2', position: { line: 8, column: 68 } }
+    });
+    expect(deps[4]).is.eql({
+      name: { value: 'github.com/pmezard/go-difflib', position: { line: 0, column: 0 } },
+      version: { value: 'v1.3.0', position: { line: 16, column: 41 } }
+    });
+  });
+
   it('tests go.mod with a module in import', async () => {
```

Comment:
One more test case
1. Two different modules pointing to same replace with module. (your should not find two deps entry in the request for same module).